### PR TITLE
re-enable Pylint on a number of view files

### PR DESCRIPTION
**Pull request by willbarrett**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [X] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Re-enables pylint checking on a number of view files.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@john-bodley @dpgaspar @mistercrunch

### CODE REVIEW

**Review by dpgaspar — APPROVED** [review]

File: superset/views/datasource.py [hunk 1]
```diff
@@ -100,8 +98,8 @@ def external_metadata(self, datasource_type=None, datasource_id=None):
             database = (
                 db.session.query(Database).filter_by(id=request.args.get("db_id")).one()
             )
-            Table = ConnectorRegistry.sources["table"]
-            datasource = Table(
+            table = ConnectorRegistry.sources["table"]
```
Comment (mistercrunch) [inline superset/views/datasource.py]: mmmh, this is actually a pointer to a class name, I'd either stick with `Table` or call it `table_cls` or equivalent

File: superset/views/sql_lab.py [hunk 2]
```diff
@@ -126,11 +127,11 @@ class SavedQueryView(SupersetModelView, DeleteMixin):
 
     show_template = "superset/models/savedquery/show.html"
 
-    def pre_add(self, obj):
-        obj.user = g.user
+    def pre_add(self, item):
```
Comment (mistercrunch) [inline superset/views/sql_lab.py]: NIT: `item` is probably better, but `obj` is not outrageous, we could add `obj` to Pylint's config of allowed object names.

File: superset/views/sql_lab.py [hunk 3]
```diff
@@ -176,13 +177,14 @@ def show(self, pk):
...
-        return db.session.query(TabState.user_id).filter_by(id=tab_state_id).scalar()
+def _get_owner_id(tab_state_id):
+    return db.session.query(TabState.user_id).filter_by(id=tab_state_id).scalar()
+
 
+class TabStateView(BaseSupersetView):
     @has_access_api
     @expose("/", methods=["POST"])
-    def post(self):
+    def post(self):  # pylint: disable=no-self-use
```
Comment (mistercrunch) [inline superset/views/sql_lab.py]: That rule is pretty annoying. I wouldn't mind turning it off.

File: superset/views/sql_lab.py (hunk 2 above)
Comment (dpgaspar) [reply superset/views/sql_lab.py]: I think that the problem is related with the parent's signature, on FAB the method is `pre_add(self, item)`

File: superset/views/datasource.py (hunk 1 above)
Comment (dpgaspar) [reply superset/views/datasource.py]: I'm actually starting to use `table_cls` style, so +1 on that

File: superset/views/sql_lab.py (hunk 2 above)
Comment (willbarrett) [reply superset/views/sql_lab.py]: +1 - the issue is the difference in signature.

File: superset/views/sql_lab.py (hunk 3 above)
Comment (willbarrett) [reply superset/views/sql_lab.py]: I think it's valuable outside of this context. As I was doing this work, I identified quite a few methods that could be refactored out of a large class into a stand-alone function. I'd like to keep the rule for those instances.

**Pull request merged**